jdaviz/app.py

```diff
@@ -2031,13 +2031,19 @@ def vue_data_item_remove(self, event):
         data = self.data_collection[data_label]
         orientation_plugin = self._jdaviz_helper.plugins.get("Orientation")
         if orientation_plugin is not None:
+            from jdaviz.configs.imviz.helper import base_wcs_layer_label
             orient = orientation_plugin.orientation.selected
+            if orient == data_label:
+                orient = base_wcs_layer_label
             self._reparent_subsets(data, new_parent=orient)
         else:
             self._reparent_subsets(data)
 
-        # Make sure the data isn't loaded in any viewers
-        for viewer_id in self._viewer_store:
+        # Make sure the data isn't loaded in any viewers and isn't the selected orientation
+        for viewer_id, viewer in self._viewer_store.items():
+            if orientation_plugin is not None:
+                if viewer.state.reference_data.label == data_label:
+                    self._change_reference_data(base_wcs_layer_label, viewer_id)
```

Comment:
Should we only run this when things are WCS linked? You don't want this to run when it is pixel linked and you are deleted the first image, right?

Reply:
Hmm, I think you're right. I'll add a check on `_link_type` at 2044.

Reply:
Of course, the orientation layers aren't even visible to delete when pixel-linked, but I guess one could call `vue_data_item_remove` manually.